File: Inventario.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Other/File.java to edit this template
 */

/**
 *
 * @author Estudiante
 */
import java.util.ArrayList;

public class Inventario {

public static void main(String[] args)

{
ArrayList<String> inventario = new
ArrayList<>();
inventario.add("Espada de hierro");
inventario.add("Pocion de vida");
inventario.add("Escudo de madera");
inventario.add("Llave dorada");
System.out.println("=== INVENTARIO===");
for (int i = 0; i <inventario.size(); i++) {
System.out.println((i+1)+"."+inventario.get(1+i));

}
// for-each (mas limpio)

for (String item : inventario) {

System.out.println("- " + item);
}
        int xp = 0;
        int nivel = 1;
        int xpNecesario = 100;
        int[] batallas = {20, 35, 15, 40, 30};

        for (int xpGanado : batallas) {   // ? aquí va batallas
            xp += xpGanado;

            System.out.println("Ganaste " + xpGanado + " XP");
            System.out.println("XP actual: " + xp);

            if (xp >= xpNecesario) {      // ? aquí va xpNecesario
                nivel++;
                xp -= xpNecesario;

                System.out.println("¡Subiste a Nivel " + nivel + "!");
                System.out.println("XP restante: " + xp);
            }
        }

        System.out.println("Nivel final: " + nivel);
        System.out.println("XP final: " + xp);
    }
}

File: Inventario.py
# BLOQUE 3 - Ciclos

# Inventario

inventario = [
    'Espada de hierro',
    'Pocion de vida',
    'Escudo de madera',
    'Llave dorada'
]

print('=== INVENTARIO ===')

for i, item in enumerate(inventario, 1):
    print(f'{i}. {item}')

buscar = 'Pocion de vida'

if buscar in inventario:
    print(f'[OK] {buscar} encontrada')
else:
    print(f'[X] {buscar} no disponible')

# Combate while

vida_hero = 80
vida_enemigo = 60
ronda = 1

while vida_hero > 0 and vida_enemigo > 0:

    dano_heroe = 15
    vida_enemigo -= dano_heroe

    dano_enemigo = 10
    vida_hero -= dano_enemigo

    print(f'Ronda {ronda}: Hero={vida_hero} | Enemigo={vida_enemigo}')

    ronda += 1

resultado = 'VICTORIA!' if vida_hero > 0 else 'DERROTA'

print(resultado)

# Ejercicio 3A

xp = 0
nivel = 1
xp_necesario = 100

batallas = [20, 35, 15, 40, 30]

for xp_ganado in batallas:

    xp += xp_ganado

    print(f'XP actual: {xp}')

    if xp >= xp_necesario:
        nivel += 1
        xp -= xp_necesario

        print(f'Nivel {nivel}')
